packages/authorization-platformstate-writer/src/consumerServiceV1.ts

```diff
@@ -1,20 +1,512 @@
+        return Promise.resolve();
+      } else {
+        const clientPurposesIds = clientEntry?.clientPurposesIds || [];
+
+        const platformClientEntry: PlatformStatesClientEntry = {
+          PK: pk,
+          state: itemState.active,
+          clientKind: clientEntry.clientKind,
+          clientConsumerId: clientEntry.clientConsumerId,
+          clientPurposesIds,
+          version: msg.version,
+          updatedAt: new Date().toISOString(),
+        };
+        await upsertPlatformClientEntry(platformClientEntry, dynamoDBClient);
+
+        if (platformClientEntry.clientPurposesIds.length > 0) {
+          const map: Map<PurposeId, TokenGenerationStatesClientPurposeEntry> =
+            new Map();
+
+          for (const purposeId of clientPurposesIds) {
+            const states = await retrievePlatformStatesByPurpose(
+              purposeId,
+              dynamoDBClient
+            );
+
+            const pk = makeTokenGenerationStatesClientKidPurposePK({
+              clientId,
+              kid,
+              purposeId,
+            });
+
+            const clientKidPurposeEntry: TokenGenerationStatesClientPurposeEntry =
+              {
+                PK: pk,
+                consumerId: platformClientEntry.clientConsumerId,
+                clientKind: platformClientEntry.clientKind,
+                GSIPK_eserviceId_descriptorId: makeGSIPKEServiceIdDescriptorId({
+                  eserviceId: states.purposeEntry.purposeEserviceId,
+                  descriptorId: states.agreementEntry.agreementDescriptorId,
+                }),
+                agreementState: states.agreementEntry.state,
+                GSIPK_purposeId: purposeId,
+                purposeState: states.purposeEntry.state,
+                purposeVersionId: states.purposeEntry.purposeVersionId,
+                GSIPK_clientId: clientId,
+                GSIPK_kid: makeGSIPKKid(kid),
+                GSIPK_clientId_purposeId: makeGSIPKClientIdPurposeId({
+                  clientId,
+                  purposeId,
+                }),
+                publicKey: pem,
+                updatedAt: new Date().toISOString(),
+              };
+
+            await upsertTokenStateClientPurposeEntry(
+              clientKidPurposeEntry,
+              dynamoDBClient
+            );
+
+            map.set(purposeId, clientKidPurposeEntry);
+          }
+
+          for (const purposeId of clientPurposesIds) {
+            const secondRetrievalStates = await retrievePlatformStatesByPurpose(
+              purposeId,
+              dynamoDBClient
+            );
+            const addedClientKidPurposeEntry = map.get(purposeId);
+
+            if (!addedClientKidPurposeEntry) {
+              throw genericInternalError(
+                `Error during upsert for entry related to purpose id: ${purposeId}`
+              );
+            }
+            if (
+              secondRetrievalStates.agreementEntry.state !==
```

Comment:
same considerations [here](https://github.com/pagopa/interop-be-monorepo/pull/1062#discussion_r1819107878)

Reply:
Updated